core/src/main/java/hudson/model/Queue.java

```diff
@@ -1821,6 +1827,27 @@ public String getInQueueForString() {
         public QueueTaskFuture<Executable> getFuture() { return future; }
 
         /**
+         * Gets a {@link Task}, which is guaranteed to have the {@link ExportedBean}
+         * attribute.
+         * This method is designed to workaround JENKINS-28361
+         * @return {@link Task} instance with {@link ExportedBean} attribute.
+         *      If the original task has no such attribute, a new {@link ProxyTask}
+         *      will be created. Returns null if {@link #task} is null.
+         * @since TODO
+         */
+        @Restricted(NoExternalUse.class)
+        @Exported(name = "task")
+        public @CheckForNull Task getTaskForAPI() {
+            final Task outputTask = task;
+            if (outputTask == null) {
+                return null;
+            }
+            
+            final ExportedBean bean = outputTask.getClass().getAnnotation(ExportedBean.class);
+            return bean != null ? outputTask : new ProxyTask(outputTask);
```

Comment:
I doubt, because it completely hides the task, which may cause erroneous conclusions by API users. Such Proxy dumps some basic data at least. Let's let others to review the PR and provide their feedback on the approach.

In any case, I think the <code>ProxyTask</code> class makes some added value, so I would leave it in the code